tests/run-make/short-ice/rmake.rs

```diff
@@ -0,0 +1,41 @@
+// Backtraces in internal compiler errors used to be unbearably long, spanning
+// multiple hundreds of lines. A fix was pushed in #108938, and this test gathers
+// varied metrics on level 1 and full-level backtraces to check that the output
+// was shortened down to an appropriate length.
+// See https://github.com/rust-lang/rust/issues/107910
+
+//@ ignore-windows
+
+use run_make_support::rustc;
+
+fn main() {
+    let rust_test_1 =
+        rustc().set_backtrace_level("1").input("src/lib.rs").arg("-Ztreat-err-as-bug=1").run_fail();
+    let rust_test_2 = rustc()
+        .set_backtrace_level("full")
+        .input("src/lib.rs")
+        .arg("-Ztreat-err-as-bug=1")
+        .run_fail();
+
+    let mut rust_test_log_1 = rust_test_1.stderr_utf8();
+    rust_test_log_1.push_str(&rust_test_1.stdout_utf8());
+    let rust_test_log_1 = rust_test_log_1.as_str();
+
+    let mut rust_test_log_2 = rust_test_2.stderr_utf8();
+    rust_test_log_2.push_str(&rust_test_2.stdout_utf8());
+    let rust_test_log_2 = rust_test_log_2.as_str();
+
+    let rustc_query_count_full = count_lines_with(rust_test_log_2, "rustc_query_");
+
+    assert!(
+        rust_test_log_1.lines().count() < rust_test_log_2.lines().count()
+            && count_lines_with(rust_test_log_2, "__rust_begin_short_backtrace")
+                == count_lines_with(rust_test_log_2, "__rust_end_short_backtrace")
+            && count_lines_with(rust_test_log_1, "rustc_query_") + 5 < rustc_query_count_full
+            && rustc_query_count_full > 5
```

Comment:
Discussion: should we split this conjunction of multiple clauses into individual asserts, so that if the test does fail we have a chance to know which particular clause failed? I'm personally fine with the test still being `//@ ignore-windows`, but I would like to know more specificially about *which* clause causes the test to fail on Windows so we can leave a FIXME.

Reply:
That's a very good idea, actually! I will now divide the assert, add a try job, and we can run that to see which one fails, and then restore the ignore.